tests/foreman/maintain/test_advanced.py

```diff
@@ -0,0 +1,434 @@
+from robottelo.constants import MAINTAIN_HAMMER_YML
+
+pytestmark = pytest.mark.destructive
+
+
+# Common repositories for Satellite and Capsule
+common_repos = ['rhel-8-for-x86_64-baseos-rpms', 'rhel-8-for-x86_64-appstream-rpms']
+
+# Satellite repositories
+sat_611_repos = [
+    'satellite-6.11-for-rhel-8-x86_64-rpms',
+    'satellite-maintenance-6.11-for-rhel-8-x86_64-rpms',
+] + common_repos
+
+sat_612_repos = [
+    'satellite-6.12-for-rhel-8-x86_64-rpms',
+    'satellite-maintenance-6.12-for-rhel-8-x86_64-rpms',
+] + common_repos
+
+sat_613_repos = [
+    'satellite-6.13-for-rhel-8-x86_64-rpms',
+    'satellite-maintenance-6.13-for-rhel-8-x86_64-rpms',
+] + common_repos
+
+# Capsule repositories
+cap_611_repos = [
+    'satellite-capsule-6.11-for-rhel-8-x86_64-rpms',
+    'satellite-maintenance-6.11-for-rhel-8-x86_64-rpms',
+] + common_repos
+
+cap_612_repos = [
+    'satellite-capsule-6.12-for-rhel-8-x86_64-rpms',
+    'satellite-maintenance-6.12-for-rhel-8-x86_64-rpms',
+] + common_repos
+
+cap_613_repos = [
+    'satellite-capsule-6.13-for-rhel-8-x86_64-rpms',
+    'satellite-maintenance-6.13-for-rhel-8-x86_64-rpms',
+] + common_repos
+
+sat_repos = {
+    '6.11': sat_611_repos,
+    '6.12': sat_612_repos,
+    '6.13': sat_613_repos,
+}
+cap_repos = {
+    '6.11': cap_611_repos,
+    '6.12': cap_612_repos,
+    '6.13': cap_613_repos,
+}
+
+
+def test_positive_satellite_maintain_service_restart(sat_maintain):
+    """Restart service using advanced procedure run
+
+    :id: 64d3c78e-d602-43d6-bf77-f31135ed019e
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run service-restart
+
+    :expectedresults: Satellite service should restart.
+    """
+    result = sat_maintain.cli.Advanced.run_service_restart()
+    assert 'FAIL' not in result.stdout
+
+
+def test_positive_satellite_maintain_hammer_setup(request, sat_maintain):
+    """Hammer setup using advanced procedure
+
+    :id: 236171c0-5185-465e-9eec-e15dfefb41c3
+
+    :steps:
+        1. Change password for user to any string
+        2. Run advanced procedure run hammer-setup with wrong_password.
+        3. Verify wrong_password isn't updated in MAINTAIN_HAMMER_YML
+        4. Run advanced procedure run hammer-setup with changed password.
+        5. Verify changed password is updated in MAINTAIN_HAMMER_YML
+        6. Update admin password back to default
+        7. Verify default password is updated in MAINTAIN_HAMMER_YML
+
+    :expectedresults:
+        1. Run hammer setup with wrong password, it should fail and
+           password shouldn't be updated in MAINTAIN_HAMMER_YML
+        2. Run hammer setup with changed password, it should pass and
+           password should be updated in MAINTAIN_HAMMER_YML
+
+    :BZ: 1830355
+    """
+    default_admin_pass = settings.server.admin_password
+    result = sat_maintain.execute(
+        f'hammer -u admin -p {default_admin_pass} user update --login admin --password admin'
+    )
+    assert result.status == 0
```

Comment:
```suggestion
    assert sat_maintain.execute(
        f'hammer -u admin -p {default_admin_pass} user update --login admin --password admin'
    ).status == 0
```

Reply:
I wouldn't change as this it will be length >100